otx/algorithms/detection/tasks/inference.py

```diff
@@ -353,36 +359,47 @@ def _update_stage_module(self, stage_module):
             stage_module = module_prefix[self._train_type] + stage_module
         return stage_module
 
-    def _add_predictions_to_dataset(self, prediction_results, dataset, confidence_threshold=0.0):
+    def _get_shapes(self, all_results, width, height, confidence_threshold):
+        if self._task_type == TaskType.DETECTION:
+            shapes = self._det_add_predictions_to_dataset(all_results, width, height, confidence_threshold)
+        elif self._task_type in {
+            TaskType.INSTANCE_SEGMENTATION,
+            TaskType.ROTATED_DETECTION,
+        }:
+            shapes = self._ins_seg_add_predictions_to_dataset(all_results, width, height, confidence_threshold)
+        else:
+            raise RuntimeError(f"MPA results assignment not implemented for task: {self._task_type}")
+        return shapes
+
+    def _add_predictions_to_dataset(self, prediction_results, dataset, confidence_threshold=0.0,
+                                    process_saliency_maps=False, explain_predicted_classes=True):
         """Loop over dataset again to assign predictions. Convert from MMDetection format to OTX format."""
         for dataset_item, (all_results, feature_vector, saliency_map) in zip(dataset, prediction_results):
-            width = dataset_item.width
-            height = dataset_item.height
-
-            shapes = []
-            if self._task_type == TaskType.DETECTION:
-                shapes = self._det_add_predictions_to_dataset(all_results, width, height, confidence_threshold)
-            elif self._task_type in {
-                TaskType.INSTANCE_SEGMENTATION,
-                TaskType.ROTATED_DETECTION,
-            }:
-                shapes = self._ins_seg_add_predictions_to_dataset(all_results, width, height, confidence_threshold)
-            else:
-                raise RuntimeError(f"MPA results assignment not implemented for task: {self._task_type}")
-
+            shapes = self._get_shapes(all_results, dataset_item.width, dataset_item.height, confidence_threshold)
             dataset_item.append_annotations(shapes)
 
             if feature_vector is not None:
                 active_score = TensorEntity(name="representation_vector", numpy=feature_vector.reshape(-1))
                 dataset_item.append_metadata_item(active_score, model=self._task_environment.model)
 
             if saliency_map is not None:
+                labels = list(self._labels)
```

Comment:
Isn't self._labels a list already or it's a tuple? We can use a copy if it's a list

Reply:
Right, `self._labels` is a list. But I need to modify it. Below is a bit more justification.

If I will do `labels = self._labels` and then modify `labels` then `self._labels` would also be changed because `labels` is not a new list by refer to `self._labels`.
`list()` creates a new list that I can safely modify.

Reply:
It's ok, but I just meant that the implicit list() call looks like a type conversion, while copy() emphasizes that we are working within the same data type.

Reply:
Oh I see what you mean. Will update. Thanks!

Reply:
Yeah, let's keep it pythonic! copy = list(a) feels like a call of a copy constructor in c++.